csrc/backend_ops/ncnn/onnx2ncnn/onnx2ncnn.cpp

```diff
@@ -15,2900 +15,16 @@
-      continue;
-    }
-    if (slice->input(0) != shape->output(0)) {
-      continue;
-    }
-    if (concat->input(0) != slice->output(0)) {
-      continue;
-    }
-    if (reshape->input(0) != conv->output(0) || reshape->input(1) != concat->output(0)) {
-      continue;
-    }
-
-    // add reshape attr
-    auto result = query_shape(mutable_graph, concat, weights, shape_context);
-    if (!std::get<0>(result)) {
-      continue;
-    }
-    set_node_attr_ai(*reshape, "shape", std::get<1>(result));
-
-    // reconstruct graph
-    {
-      // remove reference
-      node_reference[reshape->input(1)] -= 1;
-      node_reference[concat->input(0)] -= 1;
-      node_reference[slice->input(0)] -= 1;
-      node_reference[shape->input(0)] -= 1;
-
-      // remove tensor/blob on edge
-      blob_names.erase(slice->input(0));
-      blob_names.erase(slice->input(1));
-      blob_names.erase(slice->input(2));
-      blob_names.erase(slice->input(3));
-      weights.erase(slice->input(1));
-      weights.erase(slice->input(2));
-      weights.erase(slice->input(3));
-
-      blob_names.erase(concat->input(0));
-      blob_names.erase(concat->input(1));
-      weights.erase(concat->input(1));
-
-      blob_names.erase(reshape->input(0));
-
-      // update edge
-      shape->clear_input();
-      reshape->clear_input();
-      reshape->add_input(conv->output(0));
-
-      shape->set_op_type("noop_reducedncnn");
-      slice->set_op_type("noop_reducedncnn");
-      concat->set_op_type("noop_reducedncnn");
-
-      reduced_node_count += 3;
-    }
-    i += 3;
-  }
-}
-
-static void fuse_binaryop_with_scalar(onnx::GraphProto* mutable_graph,
-                                      std::map<std::string, onnx::TensorProto>& weights,
-                                      std::map<std::string, int>& node_reference,
-                                      std::set<std::string>& blob_names, int& reduced_node_count) {
-  int node_count = mutable_graph->node_size();
-  for (int i = 0; i < node_count; i++) {
-    onnx::NodeProto* node = mutable_graph->mutable_node(i);
-
-    // Add/Sub/Mul/Div/Min/Max/Pow
-    if (node->op_type() == "Add" || node->op_type() == "Sub" || node->op_type() == "Mul" ||
-        node->op_type() == "Div" || node->op_type() == "Max" || node->op_type() == "Min" ||
-        node->op_type() == "Pow") {
-      if (weights.find(node->input(1)) == weights.end()) continue;
-
-      const onnx::TensorProto& scalar_b = weights[node->input(1)];
-      if (scalar_b.dims_size() != 0 || get_tensor_proto_data_size(scalar_b) != 1) continue;
-
-      float b = get_node_attr_from_input<float>(scalar_b);
-
-      node_reference[node->input(1)] -= 1;
-
-      std::string input = node->input(0);
-
-      node->clear_input();
-      node->add_input(input);
-
-      onnx::AttributeProto* attr_with_scalar = node->add_attribute();
-      attr_with_scalar->set_name("with_scalar");
-      attr_with_scalar->set_i(1);
+#include <algorithm>
```

Comment:
It's very nice to divide onnx2ncnn into several source files. Maybe we can make a PR to ncnn?

Reply:
ncnn support pnnx, no one maintains `onnx2ncnn`.

It is not a good time to make PR.